docs/topics/ingress.md

```diff
@@ -0,0 +1,226 @@
+
+Now we need to create the cluster issuer objects defined above:
+
+```
+$ kubectl create -f staging-issuer.yaml 
+clusterissuer.certmanager.k8s.io/letsencrypt-staging created
+$ kubectl create -f prod-issuer.yaml 
+clusterissuer.certmanager.k8s.io/letsencrypt-prod created
+```
+
+## Putting it all together
+
+Continuing [the example from the `cert-manager` documentation](http://docs.cert-manager.io/en/latest/tutorials/acme/quick-start/index.html), let's check that everything works, and you can create an ingress with TLS - the following will create a deployment, service, and ingress using [the Kubernetes Up and Running Demo](https://github.com/kubernetes-up-and-running/kuard). Make sure to change the domain in the ingress object to the domain you own, and that you configured to use the IP of the Nginx ingress using your DNS provider (recall how for this we used `*.kube.radu.sh`):
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: kuard
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kuard
+    spec:
+      containers:
+      - image: gcr.io/kuar-demo/kuard-amd64:1
+        imagePullPolicy: Always
+        name: kuard
+        ports:
+        - containerPort: 8080
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: kuard
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+  selector:
+    app: kuard
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: kuar-demo
+  annotations:
+    kubernetes.io/ingress.class: "nginx"    
+    certmanager.k8s.io/cluster-issuer: "letsencrypt-prod"
+    certmanager.k8s.io/acme-challenge-type: http01
+
+spec:
+  tls:
+  - hosts:
+    - kuar-demo.kube.radu.sh
+    secretName: kuar-demo-prod-tls
+  rules:
+  - host: kuar-demo.kube.radu.sh
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: kuard
+          servicePort: 80
+```
+
+In theory, this will create a `Certificate` resource in your namespace, backed by Let's Encrypt:
+
+```
+$ kubectl get certificate
+NAME                 READY   SECRET               AGE
+kuar-demo-prod-tls   True    kuar-demo-prod-tls   9m
+```
+
+At this point, the host defined in your ingress configuration should be accessible, with a TLS certificate, and pointing to the demo:
+
+<img src="img/kuard-tls.png" style="height: 500px;" />
+
+Make sure to change the email in the cluster issuer to your own, and configure the domain and host of the ingress to the ones you own.
+
+> Make sure to delete the ingress after you are done with the demo, as it may expose sensitive information about your cluster.
+
+
+## Brigade gateways and TLS ingress
+
+Now that we have everything set up, it's time to deploy Brigade. Since we want to also expose at least one gateway publicly, we will use the generic gateway, which can be used to create Brigade jobs from arbitrary event sources, through webhooks.
```

Comment:
In order for this to work, Brigade must be installed *after* the Ingress/TLS? Wouldn't It work vice versa?

Reply:
Well, this is mostly for the gateways - if you deploy the gateway before `cert-manager`, you wouldn't get TLS.